llmfoundry/utils/exceptions.py

```diff
@@ -42,9 +41,7 @@ def __init__(self, dataset_name: str, split: str,
 class UnknownExampleTypeError(KeyError):
     """Error thrown when an unknown example type is used in a task."""
 
-    def __init__(self, example: Mapping) -> None:
-        self.example = example
```

Comment:
What's another way to throw useful error messages? We'd need to make our private global constant keys public if we want to pass example as a mapping in.

Reply:
by "private global constant keys" do you mean _ALLOWED_RESPONSE_KEYS and _ALLOWED_PROMPT_KEYS ?

Reply:
Ah sorry I just meant we should keep `self.example=example` because Jane may expect this error to have an example attribute. Feel free to remove if @jjanezhang agrees to it.